build.gradle

```diff
@@ -125,93 +127,173 @@ configurations.all {
 // import dependencies
 apply from: 'dependencies.gradle'
 
-sourceSets{
+/** Output dir of the OpenJPA enhancer: @value */
+String openjpaOutputBaseDir = "$buildDir/classes/javaopenjpa"
+sourceSets {
     main {
-        java.srcDirs=['src']
-        resources.srcDirs=['src']
+        java.srcDirs = ['src']
+        resources.srcDirs = ['src']
+        description "Java source code of the OpenCms-core. It includes gwt-RCP classes."
     }
     modules {
-        java.srcDirs=['src-modules']
-        resources.srcDirs=['src-modules']
+        java.srcDirs = ['src-modules']
+        resources.srcDirs = ['src-modules']
+        description "Java source code of the OpenCms-core modules for the GWT compilation. The javac compiled classes are ignored." +
+                " The javac compilation is still valuable to identify errors / warnings in the source code." +
+                " This code is ordinary java code (no gwt-classes)."
     }
     gwt {
-        java.srcDirs=['src-gwt']
-        resources.srcDirs=['src-gwt']
+        java.srcDirs = ['src-gwt']
+        resources.srcDirs = ['src-gwt']
+        description "Java-GWT source code that will be compiled by the GWT compiler into javascript."
     }
     setup {
-        java.srcDirs=['src-setup']
-        resources.srcDirs=['src-setup']
+        java.srcDirs = ['src-setup']
+        resources.srcDirs = ['src-setup']
     }
     test {
-        java.srcDirs=['test']
-        resources.srcDirs=['test','src','src-modules','src-setup']
+        java.srcDirs = ['test']
+        resources.srcDirs = ['test', 'src', 'src-modules', 'src-setup']
     }
 
     testGwt {
-        java.srcDirs=['src-gwt','test-gwt']
-        resources.srcDirs=['src-gwt','test-gwt']
+        java.srcDirs = ['src-gwt', 'test-gwt']
+        resources.srcDirs = ['src-gwt', 'test-gwt']
     }
 }
 
 // adding generated dependencies during configuration phase to avoid build path problems in IDE
-sourceSets.modules.compileClasspath += files(sourceSets.main.output.classesDir) { builtBy 'compileJava' }
-sourceSets.setup.compileClasspath += files(sourceSets.main.output.classesDir) { builtBy 'compileJava' }
-sourceSets.gwt.compileClasspath += files(sourceSets.main.output.classesDir) { builtBy 'compileJava' }
-sourceSets.gwt.compileClasspath += files(sourceSets.modules.output.classesDir) { builtBy 'modulesClasses' }
-sourceSets.test.compileClasspath += fileTree('webapp/setup/database'){ include '**/*.jar' }
-sourceSets.test.compileClasspath += files(sourceSets.setup.output.classesDir) { builtBy 'setupClasses' }
```

Comment:
This part of the build script had already been changed. The current head is compatible with Gradle 4.x.